antlr4cpp/antlr/v4/runtime/atn/grammar_atn.cpp

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Terence Parr, Sam Harwell. Licensed under the BSD license. See LICENSE in the project root for license information.
+#include "stdafx.h"
+
+#include <stdexcept>
+
+#include "grammar_atn.hpp"
+
+namespace antlr4 {
+namespace atn {
+
+	class grammar_atn::private_data
+	{
+	};
+
+	grammar_atn::grammar_atn(atn_type grammar_type, int32_t max_token_type)
+		: _grammar_type(grammar_type)
+		, _max_token_type(max_token_type)
+		, _data(std::make_unique<private_data>())
+	{
+	}
+
+	grammar_atn::~grammar_atn()
+	{
+	}
+
+	size_t grammar_atn::context_cache_size() const
+	{
+		throw std::runtime_error("not implemented");
```

Comment:
:question: We should standardize the manner in which we create stub methods that have not yet been translated from Java to C++. This is one possibility. Any suggestions for alternatives?